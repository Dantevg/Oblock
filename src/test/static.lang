{
	var Vector
	Vector = {
		const prototype = {
			init(x, y) => {
				this.x, this.y = x, y
			}
			
			add(that) => {
				this.x = this.x + that.x
				this.y = this.y + that.y
			}
			
			toString() => "Vector (" + this.x + ", " + this.y + ")"
		}
		
		add(a, b) => {
			print(a.toString(), b.toString())
		}
		
		add(a, b) => Vector(a.x + b.x, a.y + b.y)
		
		"()" (x, y) => {
			_Proto = prototype
			var x, y = x, y
		}
	}
	
	vec = Vector(10, 20)
	print(vec.toString())
	vec.add(Vector(2, 1))
	print(Vector.add(vec, Vector(1, 2)).toString())
	print(vec.toString())
}
