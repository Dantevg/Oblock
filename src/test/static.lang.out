Vector (10, 20)
Vector (13, 23)
Vector (12, 21)
